src/core/types/object/index.js

```diff
@@ -163,19 +164,20 @@ export function objectsEqualShallow(o1, o2) {
 }
 
 /**
- * @param {T} obj
+ * @param {{[prop:string]: R|undefined}} obj
  * @param {string} prop
- * @param {function(T, string):R} factory
+ * @param {(obj:{[prop:string]:R|undefined}, str:string) => R} factory
  * @return {R}
- * @template T,R
+ *
+ * @template {any} R
```

Comment:
```suggestion
 * @template R
```

Reply:
Thats an implicit any

Reply:
Does TS flag it as one? I thought it was optional for if you wanted to constrain the type of `R`, but maybe. If it's any 
```suggestion
 *
 * @template {*} R
```

Reply:
Ah, it doesn't! I was wrong. Updating